ref: get the binaries from samora-lang official repository

EusebioSimango opened the pull request:
This PR gets directly the binary from assets of the latest release on the samora-lang's official repository, instead of installing go and then build the binary.

#### BUGs FIXED:
- Crash when using a non-debian-based distro when go is not installed, 'cuz the last version use to use `apt` to get `go`

EusebioSimango commented:
@obadiaspelembe  :eyes:

obadiaspelembe commented:
Hi @EusebioSimango,

Thanks for the PR, I'm going to review it and merge ASAP.

Look forward my feedback.

Cheers

obadiaspelembe commented:
Hi @EusebioSimango I've tested and it works.

Some notes I'd add related to the `jq`, we are assuming that the tool is pre-installed by default, although, we should add it as pre-requisite or install it via our installer.

I've tested it in a fresh SO and I had to install the `jq` before running the installer.

Why do you think about it?

EusebioSimango commented:
Hi @obadiaspelembe, well noted, as the idea is not to limit it to certain distros, I prefer to add `jq` as a prerequisite because the installation can be done in different ways for Debian and non-Debian based distros.
will work on that, and update de docs.

EusebioSimango commented:
Just added `jq` a as per-requisite, and as you can see on the commit history, also added support for macOS, and an action to test the unix installer on both Linux and MacOs.
